### DISCUSSION TITLE
I have integrated faro. when i deployed in dev or stg the collect call is going in infinite loop

### DISCUSSION THREAD

**Discussion by akhilmukkamala-phenom** [grafana/faro-web-sdk#516] [Q&A]
const faro = initializeFaro({
    url: `${window.env.FARO_URL}`,
    apiKey: `${window.env.REACT_APP_API_KEY}`,
    instrumentations: [...getWebInstrumentations(), new TracingInstrumentation()],
    app: {
        name: 'qahub-ui',
    },
});
// get OTel trace and context APIs
const { trace, context } = faro.api.getOTEL();

const tracer = trace.getTracer('default');
const span = tracer.startSpan('click');
context.with(trace.setSpan(context.active(), span), () => {
    span.end();
});

i added my code in index.tsx file. 
and these are versions 

 "@grafana/faro-web-sdk": "^1.3.7",
    "@grafana/faro-web-tracing": "^1.3.6",

**Comment by codecapitano** [comment]
Hi @akhilmukkamala-phenom 

I see you have old versions of Faro and the packages are of different versions as well—current version is 1.4.2. 
We always upgrade all Faro packages at the same time and do not guarantee that different package versions always work together.

Form the code example it's hard to say why this happens.
You can enable the Faro logger for debugging.

I would recommend to:
* upgrade all Faro packages to the most recent version
* if the problem still persists enable the logger to get more insights in what's happening.

[How to activate debugging](https://github.com/grafana/faro-web-sdk/blob/main/docs/sources/tutorials/quick-start-browser.md#how-to-activate-debugging)
